felix/rules/endpoints.go

```diff
@@ -512,3 +556,76 @@ func EndpointChainName(prefix string, ifaceName string) string {
 		MaxChainNameLength,
 	)
 }
+
+// MaxPolicyGroupUIDLength is sized for UIDs to fit into their chain names.
+const MaxPolicyGroupUIDLength = MaxChainNameLength - len(PolicyGroupInboundPrefix)
+
+// PolicyGroup represents a sequence of one or more policies extracted from
+// a list of policies.  If large enough it may be programmed into its own
+// chain.
+type PolicyGroup struct {
+	Tier        string
+	Direction   PolicyDirection
+	PolicyNames []string
+	Selector    string
+	cachedUID   string
+}
+
+func (g *PolicyGroup) UniqueID() string {
+	if g.cachedUID != "" {
+		return g.cachedUID
+	}
+
+	hash := sha3.New224()
+	write := func(s string) {
+		_, err := hash.Write([]byte(s))
+		if err != nil {
+			log.WithError(err).Panic("Failed to write to hasher")
+		}
+		_, err = hash.Write([]byte("\n"))
+		if err != nil {
+			log.WithError(err).Panic("Failed to write to hasher")
+		}
+	}
+	write(g.Tier)
+	write(g.Selector)
+	write(fmt.Sprint(g.Direction))
+	write(strconv.Itoa(len(g.PolicyNames)))
+	for _, name := range g.PolicyNames {
+		write(name)
+	}
```

Comment:
* Tier: needed (in enterprise) to disambiguate tierA/policy1 vs tierB/policy1; different policies with the same policy name.
* Direction and Selector I found useful to make the object self describing.

Tier needs to be in the hash.  Direction and selector aren't strictly needed in the hash (I think) but I was finding it easier to prove to myself that all bases were covered if I included them rather than try to prove tat it works without them  In an earlier iteration I didn't have the gi/go prefix, so the direction was needed.